### ISSUE TITLE
prometheus  - health checks

### ISSUE THREAD

**Issue by magick93** [OnFinality-io/stellar#4]
>  - What metrics will Prometeus use in this project?

You can see the metrics that are being gathered on https://github.com/stellar/stellar-core-prometheus-exporter/blob/master/stellar_core_prometheus_exporter/exporter.py

 - From issues Github you would like Prometeus to collect only metrics of permanent containers, right?

Maybe I had the chart / deployment mis-configured, but the pod was continually restarting during an initContainer due to health checks.

The use of health checks should only happen on the main container, now when initContainers are running. 

Is this possible?

**Comment by bodyasys** [comment]
Now health are checked only from main container Stellar-core.
Most likely, problem was with initContainer initPg, because it started faster than pod with PostgreSQL, but this issue was solved by **additional initContainer checkdb**.

**Comment by magick93** [comment]
No, the problem is, stellar-core will, on first run of `new-db` create schema.

On next run of same command, it throws an error - about database already existing. 

And so the container gets stuck in a loop.

**Comment by bodyasys** [comment]
Okay
I started over. I removed all changes I had made and step by step caused error from your issue, researched all elements of system, reviewed logs in all containers, and also step by step checked structure of statefulset.yaml and stellar-core image. In Entrypoint, I noticed entry.sh script, especially the first part:
```

function stellar_core_init_db() {
  if [ -z ${INITIALIZE_DB:-} ] || [ "${INITIALIZE_DB}" != "true" ]; then
    echo "Not initializing DB (set INITIALIZE_DB=true if you want to initialize it)."
    return 0
  fi
  local DB_INITIALIZED="/data/.db-initialized"
  if [ -f $DB_INITIALIZED ]; then
    echo "Core db has already been initialized."
    return 0
  fi
  echo "Initializing core db..."
  stellar-core new-db --conf /stellar-core.cfg
  echo "Finished initializing core db"
  touch $DB_INITIALIZED
}

```
I checked entry.sh script, which is in container and in stellar-core image.
The main problem was that stellar-core new-db command ran faster than Pod PostgreSQL was started.
After that, I added initContainer checkdb so that initialization did not take place before PostreSQL was ready.
Then I turned on initContainer initPG and saw that the same command was executed as when starting main container and came to conclusion that initContainer initPG is not needed to initialize database because it performs the same function as part of code in the entry.sh script.
The entry.sh script works, which eliminates need for initContainer initPG.
So, do we need to decide whether we use initContainer, Job or use Entrypoint? (I'm in favor of using Entrypoint)
